### PR TITLE
add support for sns_topic_policy

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #144 
| ❓ Documentation  | yes

## Description

add support for sns_topic_policy

modify sns_topic support so that we create sns_topic_policy for inline
policy

### CODE REVIEW

File: pkg/remote/aws/test/sns_topic_policy_multiple/aws_sns_topic_policy-arn_aws_sns_eu-west-3_526954929923_user-updates-topic-arn_aws_sns_eu-west-3_526954929923_user-updates-topic.res.golden.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+ "Typ": null,
+ "Val": null,
+ "Err": "InvalidParameter: Invalid parameter: TopicArn\n\tstatus code: 400, request id: 894d9c65-03ca-501d-bfa5-c4f0fe452458"
```
Comment (eliecharra) [inline pkg/remote/aws/test/sns_topic_policy_multiple/aws_sns_topic_policy-arn_aws_sns_eu-west-3_526954929923_user-updates-topic-arn_aws_sns_eu-west-3_526954929923_user-updates-topic.res.golden.json:4]: Is this normal ?

Comment (moadibfr) [reply pkg/remote/aws/test/sns_topic_policy_multiple/aws_sns_topic_policy-arn_aws_sns_eu-west-3_526954929923_user-updates-topic-arn_aws_sns_eu-west-3_526954929923_user-updates-topic.res.golden.json]: my bad those were failed golend files that I forgot to delete...

File: pkg/resource/aws/deserializer/sns_topic_policy_deserializer.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+func (s *SNSTopicPolicyDeserializer) HandledType() resource.ResourceType {
+	return aws.AwsSnsTopicPolicyResourceType
+}
+
+func (s SNSTopicPolicyDeserializer) Deserialize(topicList []cty.Value) ([]resource.Resource, error) {
+	policies := make([]resource.Resource, 0)
+
+	for _, value := range topicList {
+		policy, err := decodeSNSTopicPolicy(value)
```
Comment (wbeuil) [inline pkg/resource/aws/deserializer/sns_topic_policy_deserializer.go:26]: We should shadow the variable here from now on to be consistent with what @elie has done on his last PR. 

```go
for _, value := range topicList {
    value := value
    policy, err := decodeSNSTopicPolicy(value)
```

**Review by wbeuil — CHANGES_REQUESTED** [review]

File: pkg/iac/terraform/state/test/sns_topic_policy/result.golden.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ },
+ {
+  "Arn": "arn:aws:sns:us-east-1:526954929923:my-topic-with-policy",
+  "Id": "arn:aws:sns:us-east-1:526954929923:my-topic-with-policy",
+  "Policy": "{\"Id\":\"__default_policy_ID\",\"Statement\":[{\"Action\":[\"SNS:Subscribe\",\"SNS:SetTopicAttributes\",\"SNS:RemovePermission\",\"SNS:Receive\",\"SNS:Publish\",\"SNS:ListSubscriptionsByTopic\",\"SNS:GetTopicAttributes\",\"SNS:DeleteTopic\",\"SNS:AddPermission\"],\"Condition\":{\"StringEquals\":{\"AWS:SourceOwner\":[]}},\"Effect\":\"Allow\",\"Principal\":{\"AWS\":\"*\"},\"Resource\":\"arn:aws:sns:us-east-1:526954929923:my-topic-with-policy\",\"Sid\":\"__default_statement_ID\"}],\"Version\":\"2012-10-17\"}"
+ }
+]
```
[the remark this answers is not in the document]
Comment (moadibfr) [reply pkg/iac/terraform/state/test/sns_topic_policy/result.golden.json:12]: done

File: pkg/resource/aws/deserializer/sns_topic_policy_deserializer.go (hunk 2 above)
Comment (moadibfr) [reply pkg/resource/aws/deserializer/sns_topic_policy_deserializer.go:26]: done

Comment (eliecharra) [reply pkg/resource/aws/deserializer/sns_topic_policy_deserializer.go:27]: it does not seem to trigger linter error, I'm wondering if this is required there

Comment (moadibfr) [reply pkg/resource/aws/deserializer/sns_topic_policy_deserializer.go:27]: I edited it anyway

**Review by eliecharra — APPROVED** [review]

**Review by wbeuil — APPROVED** [review]

**Pull request merged**